### DISCUSSION TITLE
403 error when Uploading files

### DISCUSSION THREAD

**Discussion by masonc-dev** [filebrowser/filebrowser#2717] [Q&A] [answered]
I have filebrowser installed on a VM on my computer and I use cloudflare tunnel to access it ( i cant access from the IP) but when i try to upload a file I get a 403 error

**Comment by LemuelThomas** [comment]
I'm getting the same issue, I'm surprised there's no one really talking about this

**Comment by hideedoe** [comment, marked as answer]
for me it was permissions error on my part setting up the folder in truenas scale. once I allowed the apps permission it was fixed.

**Comment by Digitalberg** [comment]
Which folder permissions are set on the Coolify file browser?

**Comment by Digitalberg** [comment]
In coolify: 

How to Fix the 403 Error When Uploading Files in Coolify Filebrowser

If you’re using Coolify with Filebrowser, you may run into a common issue:
403 Forbidden when trying to upload files.

The application loads normally, but every upload fails. This error is almost always caused by a permissions issue on the host folder where Filebrowser is trying to write.

This guide explains why it happens and how to fix it step by step.

⸻

Why the 403 Error Happens

Filebrowser runs inside a Docker container. When using Coolify, your files directory is a mounted volume, for example:

Host: /data/filebrowser
→ Container: /srv

If the user inside the container does not have permission to write to the folder on the host machine, Filebrowser cannot upload files and shows a 403 error.

So we need to make sure the host folder is writable.

⸻

Step 1 — Identify the Correct Folder

If you know the Coolify folder path, you can skip to Step 2.
Otherwise, find it with Docker:

docker ps | grep filebrowser
docker inspect <container_id> | grep -A3 Mounts

You will see something like:

"Source": "/data/filebrowser",
"Destination": "/srv"

/data/filebrowser is the folder that needs correct permissions.

⸻

Step 2 — Fix Ownership and Permissions

On your host machine:

cd /data
ls -ld filebrowser

If it is owned by root, Filebrowser may not be able to write to it.

Most versions of Filebrowser use UID 1000 inside the container.
To fix permissions:

sudo chown -R 1000:1000 /data/filebrowser
sudo chmod -R u+rwX /data/filebrowser

If Coolify is using a different UID/GID (set via PUID/PGID), use that instead.

⸻

Step 3 — Verify Inside the Container (Optional)

Enter the container:

docker exec -it <container_id> sh

Then test writing:

touch /srv/testfile

	•	If it fails, the permission problem still exists on the host.
	•	If it works, Filebrowser now has write access.

⸻

Step 4 — Restart Filebrowser

Restart the container or restart the service from Coolify:

docker restart <container_id>

Try uploading again — it should now work.

⸻

Step 5 — If You Still Get a 403 Error

Check the Filebrowser logs:

docker logs <container_id>

Look for messages like “permission denied”.

Also ensure:
	•	You are not inside a read-only directory in Filebrowser.
	•	Your reverse proxy is not blocking uploads (though that usually shows a 413 Request Too Large).

⸻

Final Thoughts

A 403 upload error in Coolify + Filebrowser almost always comes down to incorrect folder permissions.
Once you fix the ownership and restart the container, uploads begin working normally again.

If you need help identifying the correct user ID or want a custom setup, feel free to reach out — I can help you diagnose and fix it quickly.

Qureshi.me
